plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesDialog.java

```diff
@@ -0,0 +1,88 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.ui;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.jface.databinding.dialog.TitleAreaDialogSupport;
+import org.eclipse.jface.dialogs.TitleAreaDialog;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.forms.events.ExpansionEvent;
+import org.eclipse.ui.forms.widgets.ExpandableComposite;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.ui.DeployPreferencesPanel.AdvancedSectionExpansionHandler;
+import com.google.cloud.tools.eclipse.appengine.ui.AppEngineImages;
+
+public class DeployPreferencesDialog extends TitleAreaDialog {
+
+  // if the image is smaller (e.g. 32x32, it will break the layout of the TitleAreaDialog)
+  // seems like an Eclipse/JFace bug
+  private Image titleImage = AppEngineImages.appEngine(64).createImage();
```

Comment:
Images should be created in `createContents()` to ensure it's created on the SWT thread.